### PR TITLE
Personal vocab delete

### PR DESCRIPTION
**Pull request by Tanvez**
Related to [ATGU-3162](https://jira.huit.harvard.edu/browse/ATGU-3162)
- Added delete button for personal vocab list
- Added delete route to Django
- Added test for vue and Django
- Fixed Mobile view for delete button - button trails off the screen for mobile view

### CODE REVIEW

File: hedera/api.py [hunk 1]
```diff
@@ -297,6 +301,15 @@ def post(self, request, *args, **kwargs):
...
         return JsonResponse({"data": vl.data()})
 
+    def delete(self, request, *args, **kwargs):
+        data = json.loads(request.body)
+        if "id" in data:
+            count, _ = PersonalVocabularyListEntry.objects.filter(id=data["id"]).delete()
+            if(count != 0):
+                return JsonResponse({"data": True, "id": data["id"]})
+            return HttpResponseNotFound(f"could not find vocab with id={data['id']}")
+        return HttpResponseBadRequest(f"missing 'id'")
```
Comment (dodget) [inline hedera/api.py:311]: No need for an f string here, if there is no expression/variable to evaluate.

File: hedera/tests/test_api.py [hunk 2]
```diff
@@ -25,3 +38,21 @@ def test_post_personal_vocabulary_quick_add_api(self):
...
+    def test_unsuccessful_delete_personal_vocab_entry_empty_request_body(self):
+        response = self.client.delete("/api/v1/personal_vocab_list/", json.dumps({}), content_type="application/json")
+        self.assertEqual(response.status_code, 400)
+
+    def test_unsuccessful_delete_personal_vocab_entry_id_does_not_exist(self):
+        payload = {
+            "id": 9999
+        }
+        response = self.client.delete("/api/v1/personal_vocab_list/", json.dumps(payload), content_type="application/json")
+        self.assertEqual(response.status_code, 404)
```
Comment (dodget) [inline hedera/tests/test_api.py:58]: Love it.

File: static/src/js/tests/PersonalVocab.spec.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import { shallowMount, createLocalVue, mount } from '@vue/test-utils';
```
Comment (dodget) [inline static/src/js/tests/PersonalVocab.spec.js:1]: Nice to know that component mounting functions are `@vue/test-utils`

**Review by dodget — APPROVED** [review]
Nice work 👍

File: hedera/api.py (hunk 1 above)
Comment (Tanvez) [reply hedera/api.py:311]: ahh should i update it to a json response with the error message as payload?

**Review by arthurian — APPROVED** [review]
Looks good @Tanvez! :+1:

The only thing that might be worth revisiting is error responses to the client. Since this is a cross-cutting concern between this and some other outstanding PRs, it may be worth merging the existing approved PRs and following up on that separately.

**Pull request merged**